src/utils/objectKinds.ts

```diff
@@ -119,3 +120,16 @@ export const objectKindDescriptions = {
     WeakMap: "a WeakMap",
     WeakSet: "a WeakSet"
 } as const satisfies Record<DefaultObjectKind, string>
+
+// this will only return an object kind if it's the root constructor
+// example TypeError would return undefined not 'Error'
+export const getExactObjectKind = (
+    data: unknown
+): DefaultObjectKind | undefined => {
+    const constructorName: string | undefined = Object(data).name
+    return constructorName &&
+        isKeyOf(constructorName, defaultObjectKinds) &&
+        defaultObjectKinds[constructorName] === data
```

Comment:
Based on this comparison, this can't be called `data` because for this to ever pass, it would have to be the constructor itself we were checking.

I'd say rename the input parameter to constructor and rename the function to `getExactConstructorObjectKind`